### PR TITLE
Add snippet for Collection Group Query

### PR DESCRIPTION
**Pull request by BenWhitehead**
* Bump `google-cloud-firestore` to 1.1.0 which includes
  the new functionality

### CODE REVIEW

**Comment by BenWhitehead** [comment]
@crwilcox Can I get a review?

File: firestore/cloud-client/snippets.py [hunk 1]
```diff
@@ -815,3 +816,63 @@ def delete_collection(coll_ref, batch_size):
     # [END delete_full_collection]
 
     delete_collection(db.collection(u'cities'), 10)
+
+
+@pytest.mark.skip(reason="Dependant on a composite index being created,"
```
Comment (kurtisvg) [inline firestore/cloud-client/snippets.py]: Why does this need to be here? Aren't the tests in snippet_tests.py?

This should snippet should also probably have tests on it.

File: firestore/cloud-client/snippets.py [hunk 2]
```diff
@@ -815,3 +816,63 @@ def delete_collection(coll_ref, batch_size):
...
+    })
+    cities.document(u'DC').collection(u'landmarks').document().set({
+        u'name': u'National Museum of Nature and Science',
+        u'type': u'museum'
+    })
+    cities.document(u'BJ').collection(u'landmarks').document().set({
+        u'name': u'Jingshan Park',
+        u'type': u'park'
+    })
+    cities.document(u'BJ').collection(u'landmarks').document().set({
```
Comment (frankyn) [inline firestore/cloud-client/snippets.py]: Could you break apart the document ref? It's longer than I'd expect it to be.

```
bj_collection = cities.document(u'BJ').collection(u'landmarks')
bj_collection.document().set( {
  u'name': u'Beijing Ancient Observatory',
  u'type': u'museum'
}
```

Comment (BenWhitehead) [reply firestore/cloud-client/snippets.py]: done.

File: firestore/cloud-client/snippets.py (hunk 1 above)
Comment (BenWhitehead) [reply firestore/cloud-client/snippets.py]: You're right, I moved it over to the tests file and added an assertion there.

File: firestore/cloud-client/snippets_test.py [hunk 3]
```diff
@@ -255,3 +255,16 @@ def test_listen_for_changes(capsys):
 
 def test_delete_full_collection():
     snippets.delete_full_collection()
+
+
+@pytest.mark.skip(reason="Dependant on a composite index being created,"
```
Comment (kurtisvg) [inline firestore/cloud-client/snippets_test.py]: I'm still not sure why a skip is being used here. Why can't we integrate the creation of the index into the tests? 

More so, why doesn't the snippet show users how to do these things? Is that information found in a different snippet?

Comment (frankyn) [reply firestore/cloud-client/snippets_test.py]: +1 to @kurtisvg, if it hinders the usage of the sample then it should be included. 

If it's a timing issue given I/O starts tomorrow, then I'm okay with approving the sample as long as it's fixed afterwards.

Comment (BenWhitehead) [reply firestore/cloud-client/snippets_test.py]: Creation of the necessary composite index is something that today has to be done through the UI or the Firestore cli. After I/O I'm going to be running through each of the silver languages to make sure there are firestore admin clients and then work on using those to integrate the creation of the index into the setup of the test run.

Comment (kurtisvg) [reply firestore/cloud-client/snippets_test.py]: Would you please file a bug for this and then add it as `# TODO: b/####` to provide additional context for future readers?

Comment (BenWhitehead) [reply firestore/cloud-client/snippets_test.py]: Bug created, and TODO added.

**Comment by BenWhitehead** [comment]
@frankyn @kurtisvg Feedback incorporated, can I get your approvals?

**Pull request merged**